### PR TITLE
Wasp restructure

### PR DESCRIPTION
**Pull request by Dr-Electron**
# Description of change

This is a first draft of how I envision a Wasp restructure. Feedback is welcome.

## Type of change

- Documentation Enhancement

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [ ] I have made sure that added/changed links still work
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Dr-Electron** [comment]
Preview: https://iota-wiki-bxp10k8pw-iota1.vercel.app/

**Review by lucas-tortora — APPROVED** [review]
We should go through these docs and try to slim them down and add references where possible, ,  but this seems like a good start to me :)  Just one tiny UX note, could you make the link in the build sidebar to `Run and maintain` a link instead of a doc? It's a bit weird when you click on the sidebar and go to another section. Though perhaps it's just me who's been using docusaurus too long.

File: docs/learn/sidebars.ts [hunk 1]
```diff
@@ -39,6 +39,141 @@ module.exports = {
...
     },
+    {
+      type: 'category',
+      label: 'Smart Contracts',
+      collapsed: true,
+      items: [
+        {
+          type: 'category',
+          label: 'Core Concepts',
+          items: [
```
Comment (luca-moser) [inline docs/learn/sidebars.ts:50]: I think we should remove this category and just move all the categories/docs on level up, unless you plan to add the introduction and the core contracts categories as per the miro board.

**Review by luca-moser — APPROVED** [review]

File: docs/learn/sidebars.ts (hunk 1 above)
Comment (Dr-Electron) [reply docs/learn/sidebars.ts:50]: I thought we might add some other sections here. Core concepts is kind of an "advanced" topic. We will see in our final review

**Comment by Dr-Electron** [comment]
> We should go through these docs and try to slim them down and add references where possible, , but this seems like a good start to me :) Just one tiny UX note, could you make the link in the build sidebar to `Run and maintain` a link instead of a doc? It's a bit weird when you click on the sidebar and go to another section. Though perhaps it's just me who's been using docusaurus too long.

I know. I hate it 😆. But it should just be temporarily. Will think of a better solution in general

**Pull request merged**